### ISSUE TITLE
The text/* preview questions

### ISSUE THREAD

**Issue by Azlx** [ranger/ranger#2127]
#### Runtime Environment

- Operating system and version:  macOS Catalina 10.15.7

- Terminal emulator and version: iterm 2 3.3.12 
- Python version: 3.9.0
- Ranger version/commit: latest master
- Locale: en_US.UTF-8

#### Current Behavior
Open Ranger and press zh to open hidden files. It is normal to preview each file individually

<img width="895" alt="ranger-0" src="https://user-images.githubusercontent.com/26934642/96212095-87ab3980-0fa8-11eb-844b-4c9cb3b2d117.png">

However, if you press the "J" or "K" key to quickly scroll all the files, the preview of some files will be blocked

<img width="1260" alt="ranger-1" src="https://user-images.githubusercontent.com/26934642/96211964-46b32500-0fa8-11eb-9974-61da3eb145e0.png">

There is a preview queue in the task window, but it seems to be stuck all the time

<img width="694" alt="ranger-2" src="https://user-images.githubusercontent.com/26934642/96211970-4adf4280-0fa8-11eb-9213-bff581a9d4fc.png">

#### Steps to reproduce
1. Open Ranger and press zh to open hidden files
2. Press the "J" key to quickly scroll through all hidden files
3. Open the task window and see the preview queue stuck

**Comment by toonn** [comment]
Hmm, any of these files with a detected mimetype different from `text/*`? I haven't been able to reproduce this, using the master branch version of ranger.

**Comment by Azlx** [comment]
When I comment out all the contents of text/* in scope.sh, the preview will not be displayed when browsing hidden files

![ranger1](https://user-images.githubusercontent.com/26934642/96425926-247f0880-122f-11eb-86f7-8b4839546c8a.png)

**Comment by toonn** [comment]
Does this occur when you run `ranger --clean`?

**Comment by Azlx** [comment]
@toonn Yes, and a new phenomenon has been found. This problem is triggered by two conditions:

1. There are many hidden files

2. Press the "J" key to quickly traverse all hidden files



When I delete some hidden files, the preview queue in the task window decreases

**Comment by Azlx** [comment]
This problem should only be found on the Mac OS system, because I did not reproduce it with manjaro

**Comment by toonn** [comment]
Can you figure out a number of hidden files where it starts happening? Like, between 100 and 200, or something?

**Comment by Azlx** [comment]
About 20 or so, and not hidden files can also trigger
<img width="1262" alt="ranger-01" src="https://user-images.githubusercontent.com/26934642/96860084-e97e0e80-1494-11eb-9d50-099eadf4f40c.png">
![ranger-02](https://user-images.githubusercontent.com/26934642/96860091-eb47d200-1494-11eb-9a26-2c8013c55dbf.png)

**Comment by toonn** [comment]
Do regular text files trigger it? Font previews are fairly heavy and plists are xml so it's possible they're getting interpreted by something.

**Comment by Azlx** [comment]
The bin directory in Python 3.9 is also reproduced

Preview easy separately_ Install-3.9 is normal

Fast traversal of all files will still reappear

<img width="1249" alt="ranger-01" src="https://user-images.githubusercontent.com/26934642/96954321-bdf73480-1525-11eb-8715-86397d73aee9.png">
<img width="1266" alt="ranger-02" src="https://user-images.githubusercontent.com/26934642/96954325-c0598e80-1525-11eb-916e-1ea3da5beade.png">

**Comment by Azlx** [comment]
But if it's a large number of files with explicit suffixes, the quick preview will not reappear

For example: a large number of ". yaml" files or a large number of ". py" files
